### PR TITLE
Output alert applayer v13.3

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
https://redmine.openinfosecfoundation.org/issues/3827
https://redmine.openinfosecfoundation.org/issues/5977
https://redmine.openinfosecfoundation.org/issues/6500
https://redmine.openinfosecfoundation.org/issues/6501
https://redmine.openinfosecfoundation.org/issues/5053

Describe changes:
- Fix setup-app-layer script so that it adds app-layer metadata to alerts
- add krb5 metadata to alerts
- add ftp metadata to alerts
- add tftp metadata to alerts
- output/dns: do not add empty app-layer metadata
- app-layer: do not require probing parser as fixed patterns can be enough
- app-layer plugins

#9799 with more commits

```
SV_BRANCH=pr/1465
```
https://github.com/OISF/suricata-verify/pull/1465

Draft cc @jasonish 
I think the first commits ie https://github.com/OISF/suricata/pull/9768 should be merged first in its own PR
But that gives an overview of the POC

### CODE REVIEW

**Comment by catenacyber** [comment]
Also draft : 
Should we completely remove src/detect-mqtt-connect-password.c to have keyword registration and all done in rust only ?
Does it matter if `DETECT_AL_MQTT_CONNECT_PASSWORD` is no longer a constant ?

**Comment by catenacyber** [comment]
Replaced by #9812

File: src/util-plugin.c [hunk 1]
```diff
@@ -211,4 +212,32 @@ SCCapturePlugin *SCPluginFindCaptureByName(const char *name)
     }
     return plugin;
 }
+
+#ifdef ALPROTO_DYNAMIC_NB
+static SCAppLayerPlugin *app_layer_plugins[ALPROTO_DYNAMIC_NB];
+static size_t app_layer_plugins_nb = 0;
+#endif
+
+int SCPluginRegisterAppLayer(SCAppLayerPlugin *plugin)
```
Comment (jasonish) [inline src/util-plugin.c:221]: I'd prefer to not think about registering an app-layer as a plugin. Could app-layers not be registered the same being included or as plugins?  For plugin support we just expose the API that Suricata would already be using.

Comment (catenacyber) [reply src/util-plugin.c:221]: I do not understand your sentences here... Could you rephrase ?

(Plus I need to push a plugin example)

Comment (jasonish) [reply src/util-plugin.c:221]: Why couldn't a non-plugin AppLayer not register itself through this API? In which case its no longer a plugin API, its just the API for registering plugin.  In short, putting Plugin in here seems a little artificial.

It was a mistake in the EVE file type API which I've already started to undo.

Comment (catenacyber) [reply src/util-plugin.c:221]: Ok, so this is just about renaming `SCPluginRegisterAppLayer` to `SCRegisterDynamicAppLayer`, do I get it ?

Would we want to get rid of all the static ones ?

Comment (jasonish) [reply src/util-plugin.c:221]: > Would we want to get rid of all the static ones ?

Personally I'd think so unless there is a performance hit?

I like to think of plugin support as just exposing the APIs we already use internally.  Then there is nothing really different from a built-in or plugin, and we're constantly dog-fooding the APIs.

Comment (catenacyber) [reply src/util-plugin.c:221]: And so, should this be moved out of `util-plugin.c` to `app-layer-register.c` ?
